### PR TITLE
Handle empty work name trends

### PR DESCRIPTION
**Pull request by kani3camp**
## Summary
- Add an early-return path for empty work-name input in `UpdateWorkNameTrend`
- Save an empty non-nil ranking and current `RankedAt` without calling OpenAI
- Add a unit test for the empty-input path with a dummy API key

## Tests
- `go test ./core/workspaceapp/...`

### CODE REVIEW

File: system/core/workspaceapp/work_name_trend.go [hunk 1]
```diff
@@ -21,22 +21,39 @@ func (app *WorkspaceApp) UpdateWorkNameTrend(ctx context.Context, apiKey string)
...
+	if len(workNames) == 0 {
+		workNameTrend := repository.WorkNameTrendDoc{
+			Ranking:  []repository.WorkNameTrendRanking{},
+			RankedAt: time.Now(),
+		}
+		txErr := app.Repository.FirestoreClient().RunTransaction(ctx, func(ctx context.Context, tx *firestore.Transaction) error {
+			return app.Repository.UpdateWorkNameTrend(ctx, tx, workNameTrend)
+		})
+		if txErr != nil {
+			return fmt.Errorf("in FirestoreClient().RunTransaction(): %w", txErr)
```
[the remark this answers is not in the document]
Comment (kani3camp) [reply system/core/workspaceapp/work_name_trend.go]: 対応しました。RankedAt は app.currentTime() を使うようにし、保存も app.RunTransaction() 経由へ寄せました。検証: go test ./core/workspaceapp/...

File: system/core/workspaceapp/work_name_trend.go [hunk 2]
```diff
@@ -21,22 +21,39 @@ func (app *WorkspaceApp) UpdateWorkNameTrend(ctx context.Context, apiKey string)
...
+		}
+
+		slog.Info(utils.NameOf(app.UpdateWorkNameTrend) + " finished")
+
+		return nil
+	}
```
[the remark this answers is not in the document]
Comment (kani3camp) [reply system/core/workspaceapp/work_name_trend.go]: 対応しました。空/非空どちらも rankings を組み立てた後に共通の保存ブロックへ流す形にして、Firestore 保存処理の重複を解消しました。検証: go test ./core/workspaceapp/...

File: system/core/workspaceapp/work_name_trend_test.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	err := app.UpdateWorkNameTrend(ctx, "dummy-api-key")
+	after := time.Now()
+
+	require.NoError(t, err)
+	require.NotNil(t, savedWorkNameTrend)
+	assert.NotNil(t, savedWorkNameTrend.Ranking)
+	assert.Empty(t, savedWorkNameTrend.Ranking)
+	assert.False(t, savedWorkNameTrend.RankedAt.IsZero())
+	assert.False(t, savedWorkNameTrend.RankedAt.Before(before))
+	assert.False(t, savedWorkNameTrend.RankedAt.After(after))
```
[the remark this answers is not in the document]
Comment (kani3camp) [reply system/core/workspaceapp/work_name_trend_test.go]: 対応しました。実装を app.currentTime() に寄せたうえで、テストでは nowFunc で固定時刻を注入し、RankedAt を厳密比較する形に変更しました。検証: go test ./core/workspaceapp/...

File: system/core/workspaceapp/work_name_trend.go [hunk 4]
```diff
@@ -21,22 +21,39 @@ func (app *WorkspaceApp) UpdateWorkNameTrend(ctx context.Context, apiKey string)
...
 	var workNames []string
-	generalSeats, err := app.Repository.ReadActiveWorkNameSeats(ctx, true)
+	memberSeats, err := app.Repository.ReadActiveWorkNameSeats(ctx, true)
 	if err != nil {
 		return fmt.Errorf("in ReadActiveWorkNameSeats(): %w", err)
 	}
-	memberSeats, err := app.Repository.ReadActiveWorkNameSeats(ctx, false)
+	generalSeats, err := app.Repository.ReadActiveWorkNameSeats(ctx, false)
 	if err != nil {
 		return fmt.Errorf("in ReadActiveWorkNameSeats(): %w", err)
```
[the remark this answers is not in the document]
Comment (kani3camp) [reply system/core/workspaceapp/work_name_trend.go]: 対応しました。ReadActiveWorkNameSeats のエラーに member seats / general seats の文脈を含め、どちらの取得失敗か分かるようにしました。検証: go test ./core/workspaceapp/...

File: system/core/workspaceapp/work_name_trend.go [hunk 5]
```diff
@@ -21,125 +20,129 @@ func (app *WorkspaceApp) UpdateWorkNameTrend(ctx context.Context, apiKey string)
...
-		return fmt.Errorf("in json.Unmarshal(): %w", err)
+		type workNameTrendResponse struct {
+			Rankings []repository.WorkNameTrendRanking `json:"rankings"`
+		}
+		var result workNameTrendResponse
+		err = json.Unmarshal([]byte(resp.OutputText()), &result)
+		if err != nil {
+			return fmt.Errorf("in json.Unmarshal(): %w", err)
+		}
+		rankings = result.Rankings
```
[the remark this answers is not in the document]
Comment (kani3camp) [reply system/core/workspaceapp/work_name_trend.go]: 対応しました。保存前に rankings を non-nil 空スライスへ正規化する処理を追加し、nil と non-nil の両方を確認する単体テストも追加しました。検証: go test ./core/workspaceapp/...
